modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,262 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2018, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+
+#include "../precomp.hpp"
+#include "opencv-onnx.pb.h"
+
+#ifdef HAVE_PROTOBUF
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <string>
+#include <google/protobuf/message.h>
+#include <google/protobuf/text_format.h>
+#include <google/protobuf/io/zero_copy_stream_impl.h>
+
+#endif
+
+namespace cv {
+namespace dnn {
+CV__DNN_EXPERIMENTAL_NS_BEGIN
+
+#ifdef HAVE_PROTOBUF
+using ::google::protobuf::RepeatedField;
+using ::google::protobuf::RepeatedPtrField;
+using ::google::protobuf::Message;
+using ::google::protobuf::Descriptor;
+using ::google::protobuf::FieldDescriptor;
+using ::google::protobuf::Reflection;
+
+
+namespace
+{
+  class ONNXImporter
+  {
+      onnx::ModelProto model_proto;
+
+      std::map<std::string, Mat> getWeights(
+                                        const onnx::GraphProto& graph_proto);
+      LayerParams getLayerParams(const onnx::NodeProto& node_proto);
+
+  public:
+
+    ONNXImporter(const char *onnxFile)
+    {
+        std::fstream input(onnxFile, std::ios::in | std::ios::binary);
+        if (!model_proto.ParseFromIstream(&input)) {
+            CV_Error(Error::StsUnsupportedFormat, "Failed to parse onnx model");
+        }
+    }
+
+    void populateNet(Net dstNet);
+};
+}
+
+void releaseONNXTensor(onnx::TensorProto& tensor_proto)
+{
+    if (!tensor_proto.raw_data().empty()) {
+        delete tensor_proto.release_raw_data();
+    }
+}
+
+Mat getMatFromTensor(onnx::TensorProto& tensor_proto)
+{
+    onnx::TensorProto_DataType datatype = tensor_proto.data_type();
+    CV_Assert(!tensor_proto.raw_data().empty());
+    char* val = const_cast<char*>(tensor_proto.raw_data().c_str());
+
+    std::vector<int> sizes;
+    for (int i = 0; i < tensor_proto.dims_size(); i++) {
+            sizes.push_back(tensor_proto.dims(i));
+    }
+    Mat blob;
+    if (datatype == onnx::TensorProto_DataType_FLOAT) {
+        blob.create(sizes, CV_32FC1);
```

Comment:
`create` methods allocates a data but the following line replaces data pointer => memory leak.

Reply:
done